src/formatters/Refmt.re

```diff
@@ -0,0 +1,47 @@
+module P = Js.Promise;
+
+let register = () => {
+  let config = Vscode.Workspace.getConfiguration("merlin");
+  let refmtWidthArg = switch(Js.Nullable.toOption(config##refmt##width)) {
+  | None => ""
+  | Some(formatWidth) => "-w " ++ string_of_int(formatWidth)
+  };
+
+  Vscode.Languages.registerDocumentFormattingEditProvider(
+    {scheme: "file", language: "reason"},
+    {
+      "provideDocumentFormattingEdits": () => {
+        switch (Vscode.Window.activeTextEditor) {
+        | None => Js.Promise.resolve([||])
+        | Some(textEditor) =>
+          let id = Uuid.v4();
+          let tempFileName =
+            Node.Path.join([|
+              Node.Os.tmpdir(),
+              {j|vscode-reasonml-refmt-$id.re|j},
```

Comment:
The extension must be borrowed from the input file. For rei files, the formatting would fail otherwise

Reply:
I am trying to remember now why did I go with temp file at all